src/App.js

```diff
@@ -1,23 +1,74 @@
-import logo from './logo.svg';
+import contactsArr from './contacts.json'
+import {useState} from 'react'
 import './App.css';
 
+
+
 function App() {
+  let fiveContacts =  [...contactsArr.slice(0, 5)]
+  
+  let [contactList, setContacts] = useState(fiveContacts)
+
+  const addContact = ( () => {
+    let random = Math.floor( (Math.random() * contactsArr.length) -1)
+    
+    setContacts( (contactList)=>{
+      const newArr = contactsArr.filter((el, index) => {
+        if (index === random){
+          return true
+        } else {
+          return false
+        }
+      })
+      return contactList.concat(newArr)
+    })
+
+  })
+  
+  const sortByName = ( () => {
+    const orderName = [...contactList.sort((a, b) => a.name.localeCompare(b.name))]
+    setContacts(orderName)
```

Comment:
this one is great, it's just that you should use the setState syntax with callback and return, for example like this:
```
const sortByName = ()=>{
    setContacts((prevContacts)=>{
      let sortedContacts = [...prevContacts.sort((a, b) => a.name.localeCompare(b.name))]
      return sortedContacts
    })
  };
```